### DISCUSSION TITLE
Ideas for embedding examples

### DISCUSSION THREAD

**Discussion by jdm** [servo/servo#27756] [Ideas]
Hi everyone! I've successfully created a Rust application that uses glutin to create a window and a GL context, and I have the Servo event loop running using the libsimpleservo embedding API and rendering to the window's context. Rather than declaring mission accomplished, I'm interested in ideas for simple applications that are not just a basic web browser shell that I could use for this demo. Please share anything that comes to mind!

**Comment by BlueNebulaDev** [comment]
Good job developing the sample code to embed Servo in a Glutin window.

I believe it would be a bit of a contradiction to develop simple applications using this technological stack, because having to meddle with libsimpleservo and Glutin feels quite low-level and heavy-weight: that clashes with the idea of developing a high-level HTML+CSS+JS application on Servo.

Something to build on top of this that could be very useful is an alternative to Electron: an application that runs a JavaScript app on Servo, providing the JS app with additional APIs to work as a native desktop program (APIs to access the File System, the System Tray, manage the window and more).
